transform-plugins/src/main/java/co/cask/hydrator/plugin/Decompressor.java

```diff
@@ -0,0 +1,315 @@
+      }
+
+      String field = params[0];
+      String type = params[1].toUpperCase();
+      DecompressorType cType = DecompressorType.NONE;
+
+      switch(type) {
+        case "SNAPPY":
+          cType = DecompressorType.SNAPPY;
+          break;
+
+        case "GZIP":
+          cType = DecompressorType.GZIP;
+          break;
+
+        case "ZIP":
+          cType = DecompressorType.ZIP;
+          break;
+
+        case "NONE":
+          cType = DecompressorType.NONE;
+          break;
+
+        default:
+          throw new IllegalArgumentException("Unknown decompressor type " + type + " found in mapping " + mapping);
+      }
+      if (deCompMap.containsKey(field)) {
+        throw new IllegalArgumentException("Field " + field + " already has decompressor set. Check the mapping.");
+      } else {
+        deCompMap.put(field, cType);
+      }
+    }
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+    super.configurePipeline(pipelineConfigurer);
+    parseConfiguration(config.decompressor);
+    
+    // Check if schema specified is a valid schema or no.
+    try {
+      Schema outputSchema = Schema.parseJson(config.schema);
+      List<Field> outFields = outputSchema.getFields();
+      for (Field field : outFields) {
+        outSchemaMap.put(field.getName(), field.getSchema().getType());
+      }
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Format of schema specified is invalid. Please check the format.");
+    }
+    
+    for (Map.Entry<String, DecompressorType> entry : deCompMap.entrySet()) {
+      if (!outSchemaMap.containsKey(entry.getKey())) {
+        throw new IllegalArgumentException("Field '" + entry.getKey() + "' specified to be decompresed is not " +
+                                             "present in the output schema. Please add field '" + entry.getKey() + "'" +
+                                             "to output schema or remove it from decompress");
+      }
+      if (outSchemaMap.get(entry.getKey()) != Schema.Type.BYTES && 
+        outSchemaMap.get(entry.getKey()) != Schema.Type.STRING) {
+        throw new IllegalArgumentException("Field '" + entry.getKey() + "' should be of type BYTES or STRING in " +
+                                             "output schema.");
+      }
+    }
+  }
+  
+  @Override
+  public void initialize(TransformContext context) throws Exception {
+    super.initialize(context);
+    parseConfiguration(config.decompressor);
+    try {
+      outSchema = Schema.parseJson(config.schema);
+      List<Field> outFields = outSchema.getFields();
+      for (Field field : outFields) {
+        outSchemaMap.put(field.getName(), field.getSchema().getType());
+      }
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Format of schema specified is invalid. Please check the format.");
```

Comment:
What does a system message mean to the user ? Want error message to make sense to user.